src/IB/CIBStochasticStokesSolver.cpp

```diff
@@ -0,0 +1,677 @@
+        if (level->checkAllocated(d_wide_f_idx)) level->deallocatePatchData(d_wide_f_idx);
+    }
+
+    // Free the vectors.
+    d_x_wide.setNull();
+    d_b_wide.setNull();
+
+    return;
+} // deallocateSolverState
+
+void
+CIBStochasticStokesSolver::computeFUforRFD(Vec F_rfd, Vec U_rfd)
+{
+    Pointer<CIBStochasticMethod> sib_method_ops = d_cib_strategy;
+    double KbT = sib_method_ops->getkT();
+    double L_scale = sib_method_ops->getLScale();
+
+    int free_dofs_counter = 0;
+    std::vector<PetscInt> indices;
+    std::vector<PetscScalar> U_vec;
+    std::vector<PetscScalar> F_vec;
+    indices.reserve(d_num_rigid_parts * s_max_free_dofs);
+    U_vec.reserve(d_num_rigid_parts * s_max_free_dofs);
+    F_vec.reserve(d_num_rigid_parts * s_max_free_dofs);
+    for (unsigned int part = 0; part < d_num_rigid_parts; ++part)
+    {
+        int num_free_dofs;
+        const FRDV& solve_dofs = d_cib_strategy->getSolveRigidBodyVelocity(part, num_free_dofs);
+        if (SAMRAI_MPI::getRank() == 0)
+        {
+            if (num_free_dofs)
+            {
+                // Initialize external force and torque.
+                RDV Fr;
+                Eigen::Vector3d FU_rand, TW_rand, F_rand, T_rand, U_rand, W_rand;
+                for (int d = 0; d < NDIM; ++d)
+                {
+                    RNG::genrandn(&FU_rand(d));
+                    RNG::genrandn(&TW_rand(d));
+                }
+                F_rand = (KbT / L_scale) * FU_rand;
+                T_rand = (KbT)*TW_rand;
+                d_cib_strategy->eigenToRDV(F_rand, T_rand, Fr);
+
+                U_rand = (L_scale)*FU_rand;
+                W_rand = TW_rand;
+                RDV Ur;
+                d_cib_strategy->eigenToRDV(U_rand, W_rand, Ur);
+
+                for (int k = 0; k < s_max_free_dofs; ++k)
+                {
+                    if (solve_dofs[k])
+                    {
+                        U_vec.push_back(Ur[k]);
+                        F_vec.push_back(Fr[k]);
+                        indices.push_back(free_dofs_counter);
+                        ++free_dofs_counter;
+                    }
+                }
+            }
+        }
+    }
+
+    // Create PETSc Vecs for free DOFs.
+    const int n = free_dofs_counter;
+    if (n)
+    {
+        VecSetValues(U_rfd, n, &indices[0], &U_vec[0], INSERT_VALUES);
+        VecSetValues(F_rfd, n, &indices[0], &F_vec[0], INSERT_VALUES);
+    }
+
+    VecAssemblyBegin(U_rfd);
+    VecAssemblyEnd(U_rfd);
+    VecAssemblyBegin(F_rfd);
+    VecAssemblyEnd(F_rfd);
+
+    return;
+} // computeRFDforcesAndDisplacements (Brennan Sprinkle)
+
+void
+CIBStochasticStokesSolver::ComputeRFD_RHS(Vec b, Vec y)
+{
+    Pointer<CIBStochasticMethod> sib_method_ops = d_cib_strategy;
+
+    // Get some constants
+    static const double gamma = d_sp_solver->getSpreadScale();
+    static const double beta = d_sp_solver->getInterpScale();
```

Comment:
Why are these `static`?

Reply:
I don't know but they shouldn't be. Will fix.